src/main/java/org/jenkinsci/main/modules/sshd/PublicKeyAuthenticatorImpl.java

```diff
@@ -20,27 +24,40 @@
     private final PublicKeySignatureWriter signatureWriter = new PublicKeySignatureWriter();
 
     public boolean authenticate(String username, PublicKey key, ServerSession session) {
+        boolean authenticated = this.processAuthenticate(username, key);
+
+        if(authenticated){
+            UserDetails userDetails = Jenkins.getInstance().getSecurityRealm().loadUserByUsername(username);
+            SecurityListener.fireAuthenticated(userDetails);
+        }else{
+            SecurityListener.fireFailedToAuthenticate(username);
+        }
+
+        return authenticated;
+    }
+    
+    private boolean processAuthenticate(String username, PublicKey key) {
         LOGGER.fine("Authentication attempted from "+username+" with "+key);
-        User u = User.get(username, false);
+        User u = User.get(username, false, Collections.emptyMap());
         if (u==null) {
             LOGGER.fine("No such user exists: "+username);
             return false;
         }
-
+    
```

Comment:
Please configure your editor to not introduce gratuitous whitespace changes, and revert these.

Reply:
whitespaces removed and normally IDE configured